src/components/cardsList.tsx

```diff
@@ -10,11 +12,13 @@ interface ListProps {
 
 export const CardsList: React.FC<ListProps> = ({ results, onClick }): ReactNode => {
   const navigate = useNavigate();
+  const selectedIds = useCardStore((s) => s.selectedIds);
```

Comment:
It is better to use selector here and everywhere. You can read about it https://habr.com/ru/articles/564004/.
It should be smth like that
```suggestion
  const selectedIds = useCardStore(selectedCards);
```
and in the store you should add 
```
export const selectedCards = (s) => s.selectedIds;
```

Reply:
ok!